File: w09/d01/alex_hint/mail_items/spec/mail_item_spec.rb
# MailItem should have a ship_to_address
# MailItem should have a stamp_value

require_relative 'spec_helper'
require_relative '../lib/mail_item'

describe MailItem do

  it 'should have a ship to address' do
    mailitem = MailItem.new
    mailitem.ship_to_address = "224 Main St"
    expect(mailitem.ship_to_address).to eq "224 Main St"
  end

  it "should have a stamp value" do
    mailitem = MailItem.new
    mailitem.stamp_value = "5"
    expect(mailitem.stamp_value).to eq "5"
  end

end

File: w09/d03/jimmy_davis/mail_objects_app/spec/javascripts/mailitem_spec.js
describe("a mail item", function(){

  var mailitem;

  beforeEach( function() {
    mailitem = new MailItem("902 Broadway");
  });

  it("has a ship to address", function() {
    mailitem.shipToAddress("902 Broadway");
    expect(mailitem.shipToAddress).toEqual("902 Broadway");
    var test = function() {
      mailitem.shipToAddress(902);
    };
    expect(test).toThrow();
  });

  it("has a stamp value", function() {
    mailitem.stampValue(43);
    expect(mailitem.stampValue).toEqual(43);
    var test = function() {
      mailitem.stampValue("nan");
    };
    expect(test).toThrow();
  });
});
